### ISSUE TITLE
ZigBee network map smart plugs incorrect

### ISSUE THREAD

**Issue by tombadog99** [asantaga/wiserHomeAssistantPlatform#274]
Love the new network map feature - really helpful.
Only noticed one thing that wasn't quite right. All the smart plugs say they are connected directly to the hub but that isn't the case in my setup as most of mine are daisy chained.

**Comment by msp1974** [comment]
The connections come from the parent node id the device is showing.  I assume on the signal sensor for your plugs they also show connected to the hub directly in the attributes?

Are you able to provide a json output using the wiser output json service to export to files so i can see if it is just how the hub works or there is some bug in our api.  Thx

**Comment by tombadog99** [comment]
> The connections come from the parent node id the device is showing. I assume on the signal sensor for your plugs they also show connected to the hub directly in the attributes?
> 
> Are you able to provide a json output using the wiser output json service to export to files so i can see if it is just how the hub works or there is some bug in our api. Thx

Thinking back - yes, I believe the plugs have always shown up as being connected to the hub directly even though they aren't. Always made it difficult to establish where the weak point in the network was in the past as the smart plugs are in an environment with many 1m thick stone walls which can be very challenging. 

JSONs attached

[domain-20220710-201430.json.txt](https://github.com/asantaga/wiserHomeAssistantPlatform/files/9079716/domain-20220710-201430.json.txt)
[network-20220710-201430.json.txt](https://github.com/asantaga/wiserHomeAssistantPlatform/files/9079717/network-20220710-201430.json.txt)
[schedules-20220710-201430.json.txt](https://github.com/asantaga/wiserHomeAssistantPlatform/files/9079718/schedules-20220710-201430.json.txt)
d

**Comment by msp1974** [comment]
I did look at mine before you sent this and smartplugs do not show a parent node id like other devices do, so we just set it to 0 in the api which is the node id for the hub.  Im sure they used to have this and also used to show a signal strength % but not just show online.  Sorry, not much we can do about this.  Your json files show the same thing.

I will pick up with Angelo to see if we can get an understanding from Wiser why this was changed and if it could be changed back in a future hub firmware update.

**Comment by tombadog99** [comment]
No worries. Still love the network map though!

**Comment by asantaga** [comment]
I was looking at this and alas I only have one plug..  I guess its still showing this?

**Comment by asantaga** [comment]
@tombadog99  Is this still an issue? ok to close?

**Comment by asantaga** [comment]
I'll do some more digging , ignore my ignorance but how do you know the plugs are daisy chained?

**Comment by msp1974** [comment]
This needs raising with Jamie as it is the hub that no longer supplies this info

**Comment by tombadog99** [comment]
> I'll do some more digging , ignore my ignorance but how do you know the plugs are daisy chained?

My hub is at one end of the house. Due to the construction of the house (nearly 1m thick stone walls), it is impossible that the smart plugs are all connected to the hub. The signal can barely get through one!

**Comment by asantaga** [comment]
Email sent to Jaimee, i dont think this will get fixed..  Unless the hub exposes the data we cant report on it.

**Comment by freshfieldx** [comment]
I have this setup and if I turn my middle SP off all my distant TRV go offline as they were connected to my end of the line SP. Hopefully that made sense!? 😁

**Comment by asantaga** [comment]
yes, im speaking to Jamie about this. Alas given nothing is "broken" , its just a nice to have , i dont think we can give this high priority (unlike the trvs not closing or the hub going off line etc)..

**Comment by tombadog99** [comment]
Would agree that it isn't high priority. Would be useful but not vital.
Other bugs you mention much more important.

**Comment by asantaga** [comment]
Ive not heard back from our wiser "friend", alas, this is what the hub is reporting so I think we're stuck with it..

**Comment by tombadog99** [comment]
No problem. It's still useful as it is.

**Comment by msp1974** [comment]
Will close this issue as cannot resolve without Wiser making hub firmware change.
